test/config-luatest/gh_10304_allow_specific_function_access_test.lua

```diff
@@ -0,0 +1,70 @@
+local fun = require('fun')
+local yaml = require('yaml')
+local t = require('luatest')
+local treegen = require('test.treegen')
+local server = require('test.luatest_helpers.server')
+local helpers = require('test.config-luatest.helpers')
+
+local g = helpers.group()
+
+g.test_lua_call_func = function(g)
+    local dir = treegen.prepare_directory(g, {}, {})
+    local config = {
+        credentials = {
+            users = {
+                guest = {
+                    roles = {'super'},
+                },
+                alice = {
+                    password = "ALICE",
+                    privileges = {
+                        {
+                            permissions = {"execute"},
+                            lua_call = {"foo", "bar"}
+                        },
+                    }
+                },
+            },
+        },
+        iproto = {
+            listen = {{uri = 'unix/:./{{ instance_name }}.iproto'}},
+        },
+        groups = {
+            ['group-001'] = {
+                replicasets = {
+                    ['replicaset-001'] = {
+                        instances = {
+                            ['instance-001'] = {},
+                        },
+                    },
+                },
+            },
+        },
+    }
+
+    local config_file = treegen.write_script(dir, 'config.yaml',
+                                             yaml.encode(config))
+    local opts = {config_file = config_file, chdir = dir}
+    g.server = server:new(fun.chain(opts, {alias = 'instance-001'}):tomap())
+    g.server:start()
+    g.server:exec(function()
+        local netbox = require('net.box')
+        box.cfg{listen=3303}
+        local con = netbox.connect(3303, {user='alice', password='ALICE'})
+
+        rawset(_G, 'foo', function() return true end)
+        rawset(_G, 'bar', function() return true end)
+        rawset(_G, 'baz', function() return true end)
+
+        -- User `alice` should be able to call functions `foo` and `bar`
+        -- and nothing else.
+        t.assert(con:call('foo'))
+        t.assert(con:call('bar'))
+
+        local msg = 'Execute access to function \'baz\' is denied ' ..
+            'for user \'alice\''
+        local _, err = pcall(con.call, con, 'baz')
+        t.assert_equals(tostring(err), msg)
```

Comment:
One of the `t.assert_error_*` helpers should be more convenient.

Reply:
corrected